cloudinit/config/cc_growpart.py

```diff
@@ -139,25 +142,46 @@ def available(self):
             pass
         return False
 
+    def get_size(self, output):
+        """
+        CHANGED: resizer=sfdisk partition=1 start=2048 old: size=1021952
+                 end=1024000 new: size=2045919 end=2047967
+        CHANGED: partition=1 start=2048 old: size=1021952 end=1024000
+                 new: size=2045919,end=2047967
+        NOCHANGE: partition 1 is size 83883999. it cannot be grown
+        """
+        m = before = after = None
+        noparse = "Failed to parse growpart output: %s" % output
+        try:
+            if output.startswith('CHANGED:'):
+                m = re.match(GROWPART_CHANGED, output)
+                before = m.group('old_size')
+                after = m.group('new_size')
+            elif output.startswith('NOCHANGE:'):
+                m = re.match(r'.*is\ size\ (?P<old_size>\d+)\.', output)
+                before = after = m.group('old_size')
+            else:
+                raise ValueError(noparse)
+
+        except AttributeError:
+            raise ValueError(noparse)
+
+        return (before, after)
+
     def resize(self, diskdev, partnum, partdev):
-        before = get_size(partdev)
         try:
-            util.subp(["growpart", '--dry-run', diskdev, partnum])
+            out, err = util.subp(["growpart", '-v', diskdev, partnum])
+            LOG.debug('growpart: %s', out)
+            return self.get_size(out)
```

Comment:
These two lines will *not* raise a ProcessExecutionError, so you should remove them from this `try`.
You should just do two try/excepts

    try:
        util.subp(...)
    except util.ProcessExecutionError as e:
        if e.exit_code != 1:
            util.logexc(LOG, "failed: growpart %s %s", diskdev, partnum)

    LOG.debug("growpart output: %s", out)
    try:
        return self.get_size(out)
    except ValueError as e:
        ...
You should verify that above... but the logexc should actually log the string representation of the ProcessExecutionError, which is *better* than your message here.  As that message would have stdout included, and error case, your you would not have ever logged stdout.  The ProcessExecutionError would happen before `LOG.debug('growpart: %s')` would have ran.